tests/components/test_ring.py

```diff
@@ -0,0 +1,227 @@
+"""The tests for the Ring component."""
+import time
+import unittest
+from unittest import mock
+
+from homeassistant import setup
+import homeassistant.components.ring as ring
+
+from tests.common import get_test_home_assistant
+
+ATTRIBUTION = 'Data provided by Ring.com'
+
+VALID_CONFIG = {
+    "ring": {
+        "username": "foo",
+        "password": "bar",
+    }
+}
+
+
+def mocked_requests_get(*args, **kwargs):
+    """Mock requests.get invocations."""
+    class MockResponse:
```

Comment:
We have exactly this functionality already available as `requests_mock`. See this example: https://github.com/home-assistant/home-assistant/blob/dev/tests/util/test_location.py#L62-L66